lib/markdown_videos/services/service_base.rb

```diff
@@ -0,0 +1,124 @@
+module MarkdownVideos
+  module Services
+
+    class ServiceBase
+
+      attr_accessor :id,
+                    :alt_text,
+                    :markdown_url,
+                    :width,
+                    :height,
+                    :wrapper,
+                    :classname
+
+      def self.support(markdown_url)
+        markdown_url =~ regexp
+      end
+
+      def self.regexp
+        raise "#{self.class} class should respond to `regexp`"
+      end
+
+
+      def initialize(alt_text, markdown_url, options)
+        @alt_text = alt_text
+        @markdown_url = markdown_url
+        options.each { |option, value| self.send("#{option}=", value) }
+      end
+
+      # URL to be consumed by markup
+      def url
+        raise "#{self.class} instance should respond to `url`"
+      end
+
+      # List of authorized url parameters
+      #
+      # @return [Array<Symbol>] e.g.: [:start]
+      def url_parameters
+        []
+      end
+
+      def width
+        @width || 500
+      end
+
+      def height
+        @height || 281
+      end
+
+      # @return [String] escaped alternative text given through markdown syntax
+      def alt_text
+        CGI::escapeHTML(@alt_text)
+      end
+
+      # @return [String] video id, usually consumed by `url` method in service class
+      def id
+        @id ||= self.class.regexp.match(markdown_url)[2]
+      end
+
+
+      # @return [String] service URL to be consumed by markup
+      def service_url
+        given_url = Addressable::URI.parse(markdown_url)
```

Comment:
It's required by `bundler` when you use it. But in case you don't use bundler, then you should should require it anyway. And I required it in the [wrong place](https://github.com/capripot/markdown_videos/blob/classier_implementation/lib/markdown_videos/renderer.rb#L1).